src/Compilers/CSharp/Desktop/CommandLine/CommandLineParser.cs

```diff
@@ -1355,6 +1361,27 @@ private IEnumerable<CommandLineAnalyzerReference> ParseAnalyzers(string arg, str
             }
         }
 
+        private IEnumerable<CommandLineAnalyzerDependency> ParseAnalyzerDependencies(string arg, string value, List<Diagnostic> diagnostics)
+        {
+            if (value == null)
+            {
+                AddDiagnostic(diagnostics, ErrorCode.ERR_SwitchNeedsString, MessageID.IDS_Text.Localize(), arg);
+                yield break;
+            }
+            else if (value.Length == 0)
+            {
+                AddDiagnostic(diagnostics, ErrorCode.ERR_NoFileSpec, arg);
+                yield break;
+            }
+
+            List<string> paths = ParseSeparatedPaths(value).Where((path) => !string.IsNullOrWhiteSpace(path)).ToList();
+
+            foreach (string path in paths)
+            {
+                yield return new CommandLineAnalyzerDependency(path);
```

Comment:
We don't do that kind of validation in `CommandLineParser`, it happens later. That will be part of the later change to flesh-out the compiler support.